File: output/z_ai_glm_5/zero/python/Arquivos.py

import sys

def main():
    # Lê toda a entrada de uma vez para eficiência
    input_data = sys.stdin.read().split()
    
    if not input_data:
        return

    iterator = iter(input_data)
    
    try:
        n = int(next(iterator))
        b = int(next(iterator))
        
        files = []
        for _ in range(n):
            files.append(int(next(iterator)))
    except StopIteration:
        return

    # Ordena os tamanhos dos arquivos
    files.sort()
    
    left = 0
    right = n - 1
    folders = 0
    
    # Algoritmo de dois ponteiros (greedy)
    while left <= right:
        # Tenta emparelhar o menor (left) com o maior (right)
        if files[left] + files[right] <= b:
            left += 1  # O menor arquivo foi usado no par
        
        # O maior arquivo (right) sempre é usado (sozinho ou em par)
        right -= 1
        folders += 1
        
    print(folders)

if __name__ == "__main__":
    main()

File: output/csbc_semish_2026/z_ai_glm_5/few/cpp/Arquivos.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    long long b;
    if (!(cin >> n >> b)) return 0;

    vector<long long> files(n);
    for (int i = 0; i < n; i++) {
        cin >> files[i];
    }

    sort(files.begin(), files.end());

    int left = 0;
    int right = n - 1;
    int folders = 0;

    while (left <= right) {
        if (files[left] + files[right] <= b) {
            left++;
        }
        right--;
        folders++;
    }

    cout << folders << endl;

    return 0;
}